source/material_model/equation_of_state/thermodynamic_table_lookup.cc

```diff
@@ -91,6 +93,58 @@ namespace aspect
                 if (it == unique_phase_names.end())
                   unique_phase_names.push_back(phase_volume_column_name);
               }
+
+            // Do the same for the dominant phases
+            std::vector<std::string> phase_names_one_lookup = material_lookup[i]->get_dominant_phase_names();
+            for (const auto &phase_name : phase_names_one_lookup)
+              {
+                std::vector<std::string>::iterator it = std::find(list_of_dominant_phases.begin(),
+                                                                  list_of_dominant_phases.end(),
+                                                                  phase_name);
+
+                // Each lookup only stores the index for the individual lookup, so we have to know
+                // how to convert from the indices of the individual lookup to the index in the
+                // list_of_dominant_phases vector. Here we fill the global_index_of_lookup_phase
+                // object to contain the global indices.
+                global_index_of_lookup_phase[i].push_back(std::distance(list_of_dominant_phases.begin(), it));
+
+                if (it == list_of_dominant_phases.end())
+                  list_of_dominant_phases.push_back(phase_name);
+              }
+
+            // Make sure that either all or none of the tables have a column with the dominant phase.
+            if (any_lookup_has_dominant_phase)
+              AssertThrow(material_lookup[i]->has_dominant_phase(),
+                          ExcMessage("Some of the lookup tables you read in contain outputs for the dominant phase, "
+                                     "as indicated by the column 'phase', but in at least of of the tables you use "
+                                     "this column is missing."));
```

Comment:
You will have to move this part into a separate loop over n_material_lookups after the current loop finishes. In the current form, you could find the first two tables without phase columns, the next three with phase columns, and this check would not notice, because it only checks all future materials for a phase column.

Reply:
I think I found an even simpler way. Should be fixed now.